src/tiledb/cloud/invites.py

```diff
@@ -0,0 +1,201 @@
+from tiledb.cloud import client
+from tiledb.cloud import rest_api
+from tiledb.cloud import tiledb_cloud_error
+from tiledb.cloud.rest_api.models.invitation_organization_join_email import (
+    InvitationOrganizationJoinEmail,
+)
+
+
+def accept_invitation(invitation_id: str) -> None:
+    """
+    Accept an invitation.
+
+    Args:
+        invitation_id (str): the ID of invitation about to be accepted.
+
+    Raises:
+        TileDBCloudError
+
+    Returns:
+        None
+    """
+    invitation_api = client.build(rest_api.InvitationApi)
+    try:
+        return invitation_api.accept_invitation(invitation=invitation_id)
+    except rest_api.ApiException as exc:
+        raise tiledb_cloud_error.check_exc(exc)
+
+
+def fetch_invitations(**filters):
+    """
+    Fetches a paginated list of invitations.
+
+    Args (filters):
+        organization (str): name or ID of organization to filter
+        array (str): name/uri of array that is url-encoded to filter
+        group (str): name or ID of group to filter
+        start (int): start time for tasks to filter by
+        end (int): end time for tasks to filter by
+        page (int): pagination offset
+        per_page (int): pagination limit
+        type (str): invitation type, \"ARRAY_SHARE\", \"JOIN_ORGANIZATION\"
+        status (str): Filter to only return \"PENDING\", \"ACCEPTED\"
+        orderby (str): sort by which field valid values include
+                       timestamp, array_name, organization_name
+
+    Raises:
+        TileDBCloudError
+
+    Returns:
+        dict: Invitations and pagination metadata.
+    """
+    invitation_api = client.build(rest_api.InvitationApi)
+    try:
+        return invitation_api.fetch_invitations(**filters)
+    except rest_api.ApiException as exc:
+        raise tiledb_cloud_error.check_exc(exc)
+
+
+def invite_to_organization(
+    organization: str, email_invite: InvitationOrganizationJoinEmail
+) -> None:
+    """
+    Sends email to multiple recipients with joining information
+    regarding an organization.
+
+    Args:
+        organization (str): name or UUID of organization.
+        email_invite (InvitationOrganizationJoinEmail): list of email recipients
+
+    Raises:
+        TileDBCloudError
+
+    Returns:
+        None
+    """
+    invitation_api = client.build(rest_api.InvitationApi)
+    try:
+        return invitation_api.join_organization(organization, email_invite)
```

Comment:
I don’t think we want people to have to be constructing their own internal structures to call this API. Instead, I recommend taking in a sequence of invitees (email addresses or usernames) and the role to provide them. I also don’t feel like there is any obvious order that the parameters should go in (you could easily think that either who is being invited, or the org to be invited to should go first), and because of that it might be best to make them kwargs-only, so that users can’t accidentally mess it up.

This might look like:

```python
def invite_to_organization(
    *,
    organization: str,
    recipients: Sequence[str],
    role: str,
) -> None:
    ...
```